modules/text-area/react/spec/TextArea.spec.tsx

```diff
@@ -1,44 +1,67 @@
+describe('Text Area', () => {
   const cb = jest.fn();
   afterEach(() => {
     cb.mockReset();
   });
-  test('TextArea should spread extra props', () => {
-    const component = mount(<TextArea data-propspread="test" onChange={cb} />);
-    const container = component.at(0).getDOMNode();
-    expect(container.getAttribute('data-propspread')).toBe('test');
-    component.unmount();
+
+  describe('when rendered with a placeholder', () => {
+    it('should render a text area with placeholder', () => {
+      const {getByRole} = render(<TextArea onChange={cb} placeholder={placeholder} />);
+      expect(getByRole('textbox')).toHaveAttribute('placeholder', placeholder);
+    });
   });
-});
 
-describe('Text Area Accessibility', () => {
-  const cb = jest.fn();
-  afterEach(() => {
-    cb.mockReset();
+  describe('when rendered with an id', () => {
+    it('should render a text area with a id', () => {
+      const {getByRole} = render(<TextArea id={id} onChange={cb} />);
+      expect(getByRole('textbox')).toHaveAttribute('id', id);
+    });
+  });
+
+  describe('when rendered with a value', () => {
+    it('should render a text area with a value', () => {
+      const {getByDisplayValue} = render(<TextArea onChange={cb} value={value} />);
+      expect(getByDisplayValue(value)).toBeDefined();
+    });
   });
-  test('Text Area in a FormField should pass axe DOM accessibility guidelines', async () => {
-    const html = ReactDOMServer.renderToString(
-      <FormField label="My Field" inputId="my-input-field">
-        <TextArea placeholder="Placeholder" value={'Hello'} onChange={jest.fn()} />;
-      </FormField>
-    );
-    expect(await axe(html)).toHaveNoViolations();
-  });
-
-  test('Text Area with `aria-labelledby` should pass axe DOM accessibility guidelines', async () => {
-    const html = ReactDOMServer.renderToString(
-      <>
-        <label id="123">Label</label>
-        <TextArea placeholder="Placeholder" value={'Hello'} aria-labelledby="123" onChange={cb} />;
-      </>
-    );
-    expect(await axe(html)).toHaveNoViolations();
+
+  describe('when rendered with disabled attribute', () => {
+    it('should render a disabled text area', () => {
+      const {getByRole} = render(<TextArea onChange={cb} disabled={true} />);
+      expect(getByRole('textbox')).toBeDisabled();
+    });
+  });
+
+  describe('when rendered with extra, arbitrary props', () => {
+    it('should spread extra props', () => {
+      const attr = 'test';
+      const {getByRole} = render(<TextArea onChange={cb} data-propspread={attr} />);
+      expect(getByRole('textbox')).toHaveAttribute('data-propspread', attr);
+    });
+  });
+
+  describe('when provided an input ref', () => {
+    it('should set the ref to the input element', () => {
+      const ref: React.RefObject<HTMLInputElement> = React.createRef();
+      render(<TextArea inputRef={ref} id={id} />);
+      expect(ref.current).not.toBeNull();
+      expect(ref.current).toHaveAttribute('id', id);
```

Comment:
I think the following should work:
```ts
expect(await getByRole('textbox')).toEqual(ref.current)
```

Also, as a side-note, I think it is safer for Testing Library APIs to be used with the promise API since rendering can be asynchronous. The synchronous API could break if we slightly change the implementation of components or React slightly changes timing (e.g. `useEffect` vs `componentDidMount` or concurrent mode)

Reply:
@NicholasBoll Are you saying that we should aways be using `expect(await getBy*())` instead of `expect(getBy*())`?

Reply:
I know a bit more about the React Testing Library APIs. There are `get*`, `query*`, and `find*` queries.

`find*` is most likely what we want for React because of timing. `find*` returns a promise that resolves when the element is found (retrying until it is found). If there is a delay in rendering something (hooks, concurrent mode, etc), `find*` will eventually find the element and return it for testing.

So I should update my previous code snippet:

```ts
expect(await findByRole('textbox')).toEqual(ref.current)
```

Reply:
I'm not sure if `await find*` vs `get*` is strictly necessary. React Testing Library hooks into some testing internals to fix timing issues introduced with hooks. I just consider timing to be an implementation detail of the component that should not introduce failures or flakiness of tests.

Reply:
As discussed, we agreed to leave this change to a later PR because it will be an all-or-nothing change to our TL/R tests.